inlang/source-code/versioned-interfaces/language-tag/src/interface.test.ts

```diff
@@ -10,7 +10,7 @@ describe("verify language tags", () => {
 			"de",
 			"agy",
 			"en-US",
-			"zh-Latn",
+			"zh-Hans",
 			"de-agy",
```

Comment:
de-agy is invalid according to https://schneegans.de/lv/. Where are you getting the tags from?

Reply:
This drives me crazy. 

I have now converted the .NET regex format to JS to get complete other results than from this website

[see ts playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBANgQzAcwK4OQUwCoYAoJRSYBOYMAvDAPQB6AFAPwA8yJSAJgGaEAWpmDgD56mMAFoA4gCFxIQQB8AluIQBbJcvEAjMGC0dMPVHChaxIYLyVItvBAGstDuEpTgtcVAA8tG91pgCABuCABWIFoADgDu+ipQCJEJCACeWlAQqAoQyBLSAKLiAGIASjl54oXiAHIAMhUSAMIAEuIAIgUAlAr0CCRQ4nAgYdq2wJhw4sgIJkoQvApgIDogTsviYKn6AF6806ggqdl74vYODggKp-7X+-6btqfeNihdPfRMzIgo6FhCAG0AILiABaCHEOwAugBvABMABoAMwAXy69HEX0w3igP2QgJB4MhsNR7xhAAYEXC0RivhBgCQlFEoASwRDoTCACxorqMWksEiYZBKcCsokc6kKAHk8QAThJPL5mJYoUZSBZwLZkJl8phAFYEQAOFFSnVQzXis0w0ldABU-OY2OIkBFYEBOpBAHUAJpamKpYkYi3sq2I41dADUdvR4m8DqijNCxFQEEwYpDcthAEYjTyowASagAbgAUCXQJBYMRoHUkGgMDgMJQYAAiMTiACqAGUW0WaNQYKVMFFEBMYDElFBeDAp5h4HW-nPEsgYKkQKhx+qZyAZ5hoGWlFwYPQ8YvcMgCERSGAAHTVqD0e+134N8-vGAwkswGAViAgOCYDewzIPQLbPvWWAzk28wwKErgcDeLZdKWKIwJMqYfl+P7gH+AFASAIFgQuDZQSuMHLLAcFKAhSEoUAA)

```regex
/^(
  (?<grandfathered>
    (en-GB-oed|i-ami|i-bnn|i-default|i-enochian|i-hak|i-klingon|i-lux|i-mingo|i-navajo|i-pwn|i-tao|i-tay|i-tsu|sgn-BE-FR|sgn-BE-NL|sgn-CH-DE)|
    (art-lojban|cel-gaulish|no-bok|no-nyn|zh-guoyu|zh-hakka|zh-min|zh-min-nan|zh-xiang)
  )|
  (
    (?<language>[A-Za-z]{2,3})
    (-(?<extlang>[A-Za-z]{3})){0,2}
  )
  (-(?<script>[A-Za-z]{4}))?
  (-(?<region>[A-Za-z]{2}|[0-9]{3}))?
  (-(?<variant>[A-Za-z0-9]{5,8}|[0-9][A-Za-z0-9]{3}))* 
  (-(?<extension>[0-9A-WY-Za-wy-z](-[A-Za-z0-9]{2,8})+))*
  (-x(-(?<privateuse>[A-Za-z0-9]{1,8}))+)?
)$/
```

Reply:
@samuelstroschein On a second thought, wouldn't it be very beneficial to introduce something like https://github.com/wooorm/bcp-47#parsetag-options ? Otherwise, we have to control this all on our side.

Reply:
@felixhaeberle yes, we can introduce a `parseLanguageTag` function in the future. That's what the `@inlang/language-tag` package is for

Reply:
> I have now converted the .NET regex format to JS to get complete other results than from this website

I told you^^ I struggled hard. 

We need a regex. The validation should happen the moment someone types a wrong language tag into the project settings.